### PR TITLE
Upgrade to node v18

### PR DESCRIPTION
**Pull request by r-n-o**
## Summary & Motivation
Node 16 has been deprecated since September. Time to upgrade! This PR addresses #120. Because node v18 comes with its own experimental fetch API our code which tried to mock `fetch` was failing. I've opted to always use cross-fetch regardless of whether `fetch` is available in `globalThis` because cross-fetch is meant to be used for exactly this reason! It detects when native implementations are available and uses either browser or node fetch when they're available.

Having `fetch` in a known location is nice because we have a seam to mock it.

## How I Tested These Changes
Unit tests

## Did you add a changeset?

Changeset added and packages bumped!

### CODE REVIEW

File: pnpm-lock.yaml [hunk 1]
```diff
@@ -28,11 +28,11 @@ importers:
         specifier: workspace:*
         version: link:internal/jest-config
       '@types/node':
-        specifier: ^16.0.0
-        version: 16.18.12
+        specifier: ^18.0.0
+        version: 18.0.0
```
Comment (andrewkmin) [inline pnpm-lock.yaml:32]: if we want to specify a version, we could use `v18.18.2` here as well

**Review by andrewkmin — APPROVED** [review]
lgtm!

**Comment by andrewkmin** [comment]
> I've opted to always use cross-fetch regardless of whether fetch is available in globalThis because cross-fetch is meant to be used for exactly this reason! It detects when native implementations are available and uses either browser or node fetch when they're available.

thanks for adding this clarity

File: pnpm-lock.yaml (hunk 1 above)
Comment (r-n-o) [reply pnpm-lock.yaml:32]: Yep! Good call, I'll update it to match the version in `.nvmrc`.

**Pull request merged**